plumber.go

```diff
@@ -77,12 +77,30 @@ type D[T any] struct {
 	deps      []Dependency
 	listeners []func()
 	wrappers  []func(T) T
+	name      string
+}
+
+// Named creates a new named dependency
+func Named[T any](name string) *D[T] {
+	var d D[T]
+	d.Named(name)
+	return &d
+}
+
+// Named sets a name for the dependency
+func (d *D[T]) Named(name string) *D[T] {
+	d.name = name
+	return d
 }
 
 // String return names of underlaying type
 func (d *D[T]) String() string {
 	var v T
-	return reflect.TypeOf(&v).Elem().String()
+	s := reflect.TypeOf(&v).Elem().String()
+	if d.name == "" {
+		return s
+	}
+	return fmt.Sprintf("%s(%s)", s, d.name)
```

Comment:
nit: maybe the other way around
```suggestion
	return fmt.Sprintf("%s(%s)", d.name, s)
```